### ISSUE TITLE
Cannot clear cookie set on multiple subdomains

### ISSUE THREAD

**Issue by aronmgv** [cypress-io/cypress#5723]
As mentioned [here](https://github.com/cypress-io/cypress/issues/808#issuecomment-339362359) using command  `cy.clearCookie('nameOfCookie')` we should be able to delete cookie no matter what.. but apparently since then behaviour has changed..

Didn't find an official way to clear the whitelist so tried setting defaults back to undefined or empty array - no effect whatsoever..
```js
Cypress.Cookies.defaults({
    whitelist: [] / undefined
});
```

Cypress v `3.5.0`.

Any other ideas how to remove the cookie which was whitelisted before?? Thx

**Comment by jennifer-shehane** [comment]
We will need a fully reproducible example provided. We have tests for clearing cookies and likely there is something more complex involved where your issue is occuring. Provide a complete example or we cannot move forward with fixing your issue.

**Comment by MacGyver27** [comment]
Pretty straightforward problem.. simple provide a way to clear a cookie which was whitelisted before.

The login function has this inside:
```js
Cypress.Cookies.defaults({
    whitelist: ['jwt-token','csrf-token']
});
```

During the testing I need to login as different user to test its profile.. the way to do it is to clear `jwt-token` along with `csrf-token` what I am not able to perform in any way shape or form..

**Comment by bvdr** [comment]
Here is my workaround to clear cookies that are set using `whitelist` regex and set them again:
``` js
it("Clear browser cookies and localstorage", function(){
  
  //Visit URL
  const host_url = Cypress.env('host_url');
  cy.visit(host_url)

  // Empty defaults
  Cypress.Cookies.defaults({
    whitelist: []
  });

  //Clear localStrage
  cy.clearLocalStorage()

  //Clear Cookies
  cy.clearCookies()

  // Set defaults
  Cypress.Cookies.defaults({
    whitelist: /wordpress_.*|woocommerce_.*|wp_woocommerce_.*/
  })
})
```

**Comment by aelmekeev** [comment]
We see the same issue after upgrade to 3.5.0+ of Cypress (going to check 3.8.1 today and see if it will help). It's not possible to reproduce it since it's flaky.

**Comment by aelmekeev** [comment]
We were able to identify the root cause of "inconsistency" on our side. The reason was in fact that we had two cookies with the same name but different domains. `clearCookie` did remove only one of those two cookies. I'm not 100% sure that this is exact issue that @MacGyver27 is facing but this is what we've faced.

There is certainly a difference in behaviour between 3.4.1 and 3.5.0 but I'm not 100% sure if it's expected or not.

@jennifer-shehane, this is how it can be reproduced:

`./cypress.json`

```json
{
  "baseUrl": "https://dev.mysql.com",
  "integrationFolder": "cypress/integration",
  "testFiles": "**/*.js",
  "screenshotOnRunFailure": false,
  "video": false
}
```

`./cypress/integration/spec.js`

```javascript
describe('clearCookie', () => {
  it('should remove all cookies with specified name', () => {
    cy.setCookie('test_cookie', 'value1', {
      domain: 'mysql.com'
    })
    cy.setCookie('test_cookie', 'value2', {
      domain: 'dev.mysql.com'
    })
    cy.getCookie('test_cookie')
      .should('exist')
    cy.clearCookie('test_cookie')
    cy.getCookie('test_cookie')
      .should('not.exist')
  })
})
```

```bash
docker run -it -v $PWD:/e2e -w /e2e cypress/included:3.4.1
```

```
      Spec                                                Tests  Passing  Failing  Pending  Skipped 
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔ spec.js                                   221ms        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    All specs passed!                           221ms        1        1        -        -        -  
```

```bash
docker run -it -v $PWD:/e2e -w /e2e cypress/included:3.5.0
```

```
 1) clearCookie should remove all cookies with specified name:
     AssertionError: expected { Object (name, value, ...) } to not exist
      at Assertion.<anonymous> (https://dev.mysql.com/__cypress/runner/cypress_runner.js:85804:27)
      at Assertion.get (https://dev.mysql.com/__cypress/runner/cypress_runner.js:7750:37)

     Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  spec.js                                  520ms        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                     520ms        1        -        1        -        -  
```

**Comment by jennifer-shehane** [comment]
I can still reproduce the issue provided by @aelmekeev. The cookie is not removed in Cypress 3.5.0+.

`cypress.json`

```json
{
  "baseUrl": "https://dev.mysql.com"
}
```

`spec.js`

```js
it('should remove all cookies with specified name', () => {
  cy.setCookie('test_cookie', 'value1', {
    domain: 'mysql.com'
  })
  cy.setCookie('test_cookie', 'value2', {
    domain: 'dev.mysql.com'
  })

  cy.getCookie('test_cookie').should('exist').then((cookie) => {
    cy.log(cookie)
  })
  cy.clearCookie('test_cookie')
  cy.getCookie('test_cookie').should('not.exist')
})
```

<img width="468" alt="Screen Shot 2020-10-20 at 3 08 59 PM" src="https://user-images.githubusercontent.com/1271364/96561988-3cfe2a00-12e6-11eb-9995-dd99b22c712f.png">

### 3.4.1

<img width="490" alt="Screen Shot 2020-10-20 at 3 04 05 PM" src="https://user-images.githubusercontent.com/1271364/96561717-f01a5380-12e5-11eb-9be0-70ef4da8bce9.png">


### 3.5.0

<img width="571" alt="Screen Shot 2020-10-20 at 3 05 52 PM" src="https://user-images.githubusercontent.com/1271364/96561692-e7c21880-12e5-11eb-8b8e-77485d136a0d.png">
